src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx

```diff
@@ -142,3 +180,45 @@ export default function getOnboardingTasks(
     },
   ];
 }
+
+export function getMergedTasks(organization: Organization) {
+  const taskDescriptors = getOnboardingTasks(organization);
+  const serverTasks = organization.onboardingTasks;
+
+  // Map server task state (i.e. completed status) with tasks objects
+  const allTasks = taskDescriptors.map(
+    desc =>
+      ({
+        ...desc,
+        ...serverTasks.find(serverTask => serverTask.task === desc.task),
+        requisiteTasks: [],
+      } as OnboardingTask)
+  );
+
+  // Map incomplete requisiteTasks as full task objects
+  return allTasks.map(task => ({
+    ...task,
+    requisiteTasks: task.requisites
+      .map(key => allTasks.find(task2 => task2.task === key)!)
+      .filter(reqTask => reqTask.status !== 'complete'),
+  }));
+}
+
+const PulsingIndicator = styled('div')`
+  ${pulsingIndicatorStyles};
+`;
+
+const EventWaitingIndicator = styled(p => (
+  <div {...p}>
+    {t('Waiting for first event')}
+    <PulsingIndicator />
+  </div>
+))`
```

Comment:
I actually think this is a little more clear no? Since it encapsulates the `EventWaitingIndicator` into a single component. I could split out the styled div, but I know we do other things like this no?